.github/workflows/task-get-config.yml

```diff
@@ -0,0 +1,258 @@
+name: Get Single Configuration
+
+# This workflow gets configuration for a single instance
+# Used by task-test and task-build-artifact to get their specific configuration
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        description: 'Platform name (e.g., ubuntu:noble, macos, sanitizer, coverage)'
+        type: string
+        required: true
+      architecture:
+        description: 'Architecture (x86_64, aarch64)'
+        type: string
+        required: true
+    outputs:
+      config:
+        description: "Configuration for the specific platform and architecture as JSON"
+        value: ${{ jobs.get-config.outputs.config }}
+
+jobs:
+  get-config:
+    runs-on: ${{ vars.RUNS_ON || 'ubuntu-latest' }}
+    outputs:
+      config: ${{ steps.get-config.outputs.config }}
+    steps:
+      - name: Get configuration for platform and architecture
+        id: get-config
+        shell: python
+        run: |
+          import json
+          import os
+          import sys
+
+          platform = "${{ inputs.platform }}"
+          architecture = "${{ inputs.architecture }}"
+
+          print(f"Getting configuration for platform: {platform}, architecture: {architecture}")
+
+          # Initialize default values
+          config = {
+              'env': '',
+              'container': '',
+              'pre-steps-script': '',
```

Comment:
Must we pass all 4 parameters? Can't we pass only non-default values and control the default elsewhere?
Or is this the single point where we get this info from, and we should set the defaults here?